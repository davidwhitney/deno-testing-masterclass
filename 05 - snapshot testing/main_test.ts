import { assertSnapshot } from "jsr:@std/testing/snapshot";

Deno.test("isSnapshotMatch", async (t) => {
  const a = {
    hello: "world!",
    example: 124,
  };
  await assertSnapshot(t, a);
});


/*

We also support snapshot testing out of the box.
Useful for asserting on all of your complicated object models, like this one!

To establish a snapshot you run

deno test --allow-read --allow-write -- --update

This will create a snapshot file in the `__snapshots__` directory that you should check in as your chiseled reference for the test.
Then you can run the test again without the `--update` flag to check if the snapshot matches.

deno test --allow-read


It's probably just easier to run deno test -A.
*/
