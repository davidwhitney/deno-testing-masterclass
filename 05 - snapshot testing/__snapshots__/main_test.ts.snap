export const snapshot = {};

snapshot[`isSnapshotMatch 1`] = `
{
  example: 124,
  hello: "world!",
}
`;
